dask/dataframe/shuffle.py

```diff
@@ -984,6 +984,14 @@ def fix_overlap(ddf, mins, maxes, lens):
     name = "fix-overlap-" + tokenize(ddf, mins, maxes, lens)
 
     non_empties = [i for i, l in enumerate(lens) if l != 0]
+
+    # If all empty, collapse into one partition
+    if len(non_empties) == 0:
+        divisions = (None, None)
+        dsk = {(name, 0): (ddf._name, 0)}
+        graph = HighLevelGraph.from_collections(name, dsk, dependencies=[ddf])
```

Comment:
This causes computing the output to depend on computing the first partition of the input, which may be expensive (even if it's empty). Since we know the schema, a different approach would be to drop the upstream dependencies and construct a new graph that just returns `ddf._meta`. I *think* this approach is correct, but am not sure if it matters - what you have here is fine too.

Reply:
Yeah you are right. I am also not sure how important that is but maybe it would be more important if we had a way too get min, max, len without actually loading the data. I'll see if I can make that change.